metricbeat/mb/module/wrapper.go

```diff
@@ -70,7 +70,10 @@ type metricSetWrapper struct {
 	module *Wrapper // Parent Module.
 	stats  *stats   // stats for this MetricSet.
 
-	periodic bool // Set to true if this metricset is a periodic fetcher
+	periodic         bool // Set to true if this metricset is a periodic fetcher
+	failureThreshold int  // threshold of consecutive errors needed to set the stream as degraded
```

Comment:
assuming a negativeThreshold doesn't have any meaning I'd suggest this and consecutiveErrors be an unsigned int type.

Reply:
Changed to unsigned int in df01e8ea4bd4e255575750a1a42f5a8177dafd44 and tweaked defaults and threshold evaluation